subcommands/subcommands.go

```diff
@@ -0,0 +1,97 @@
+package subcommands
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/joyent/containerpilot/client"
+	"github.com/joyent/containerpilot/config"
+)
+
+// Subcommand provides a simple object for storing a configured HTTPClient.
+type Subcommand struct {
+	client *client.HTTPClient
+}
+
+// Init initializes the configuration of a Subcommand function and the
+// HTTPClient which they utilize for control plane interaction.
+func Init(configFlag string) (*Subcommand, error) {
+	var socketPath = "/var/run/containerpilot.socket"
+
+	if configFlag != "" {
```

Comment:
Nice catch... for whatever reason early on I was receiving the `-config` flag as required even when I knew the `cfg.Control.SocketPath` had a default value. As you suggested, I was able to remove the conditional, `socketPath` crap, and only perform....

```go
cfg, err := config.LoadConfig(configFlag)
if err != nil {
        return nil, err
}
```